hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1033,26 +1075,31 @@ private long generateDiffReport(
             throw new IllegalStateException(
                 "Old and new key name both are null");
           } else if (oldKeyName == null) { // Key Created.
-            String key = codecRegistry.asObject(newKeyName, String.class);
+            String key = resolveAbsolutePath(isFSOBucket, newParentIdPathMap,
```

Comment:
Shouldn't you be doing `resolveAbsolutePath` outside if/else and compare the resolved name? I think renamed of `/vol/bucket/dir1/dir2/key` to `/vol/bucket/dir3/dir2/key` will come as Modify diff because `objectId` would be same as well as `keyName` but it should be `Rename`.

Reply:
For FSO it would be parentId/KeyName. for non fso it would be the complete path. So it would considered as a rename operation. If you look into the addToObjectIdMap I have changed the key to map, it is not just the keyname anymore